File: epoxy-processor/src/main/java/com/airbnb/epoxy/PackageConfigSettings.java
package com.airbnb.epoxy;

/**
 * Stores configuration settings for a package.
 */
class PackageConfigSettings {

  final boolean requireHashCode;
  final boolean requireAbstractModels;
  final boolean implicitlyAddAutoModels;

  private PackageConfigSettings(boolean requireHashCode, boolean requireAbstractModels,
      boolean implicitlyAddAutoModels) {
    this.requireHashCode = requireHashCode;
    this.requireAbstractModels = requireAbstractModels;
    this.implicitlyAddAutoModels = implicitlyAddAutoModels;
  }

  static PackageConfigSettings forDefaults() {
    return new PackageConfigSettings(
        PackageEpoxyConfig.REQUIRE_HASHCODE_DEFAULT,
        PackageEpoxyConfig.REQUIRE_ABSTRACT_MODELS_DEFAULT,
        PackageEpoxyConfig.IMPLICITLY_ADD_AUTO_MODELS_DEFAULT
    );
  }

  static PackageConfigSettings create(PackageEpoxyConfig configAnnotation) {
    return new PackageConfigSettings(
        configAnnotation.requireHashCode(),
        configAnnotation.requireAbstractModels(),
        configAnnotation.implicitlyAddAutoModels());
  }
}

File: epoxy-processor/src/main/java/com/airbnb/epoxy/processor/PackageConfigSettings.kt
package com.airbnb.epoxy.processor

import androidx.room.compiler.processing.XAnnotationBox
import com.airbnb.epoxy.PackageEpoxyConfig

/**
 * Stores configuration settings for a package.
 */
class PackageConfigSettings private constructor(
    val requireHashCode: Boolean,
    val requireAbstractModels: Boolean,
    val implicitlyAddAutoModels: Boolean
) {

    companion object {

        fun forDefaults() = PackageConfigSettings(
            PackageEpoxyConfig.REQUIRE_HASHCODE_DEFAULT,
            PackageEpoxyConfig.REQUIRE_ABSTRACT_MODELS_DEFAULT,
            PackageEpoxyConfig.IMPLICITLY_ADD_AUTO_MODELS_DEFAULT
        )

        fun create(configAnnotation: XAnnotationBox<PackageEpoxyConfig>) = PackageConfigSettings(
            configAnnotation.value.requireHashCode,
            configAnnotation.value.requireAbstractModels,
            configAnnotation.value.implicitlyAddAutoModels
        )
    }
}
